src/main/java/life/qbic/App.java

```diff
@@ -40,17 +41,23 @@ private static QbicDataDownloader loginToOpenBIS(
       PostmanCommandLineOptions commandLineParameters) {
 
     String password;
-    if (!commandLineParameters.passwordEnvVariable.isEmpty()) {
-
-      password = OpenBISPasswordParser.readPasswordFromEnvVariable(commandLineParameters.passwordEnvVariable, commandLineParameters.user);
-
+    if (OpenBISPasswordParser.isNotNullOrEmpty(commandLineParameters.passwordEnvVariable)) {
+
+      Optional<String> envPassword = OpenBISPasswordParser.readPasswordFromEnvVariable(commandLineParameters.passwordEnvVariable);
+      if (envPassword.isPresent()){
+        password = envPassword.get();
+      }
+      else{
+        System.out.println("No environment Variable named " + commandLineParameters.passwordEnvVariable + " was found");
+        LOG.info(String.format("Please provide a password for user '%s':", commandLineParameters.user));
+        password = OpenBISPasswordParser.readPasswordFromConsole();
+
+     }
     }
     else {
```

Comment:
```suggestion
      } else {
```

Reply:
There is no second else case, is it possible that your app.java class is not the latest version? It also has a outdated field next to it.

Reply:
yes very much possible 😄 sorry for the confusion